test-cypress/integration/explore/class.hierarchy.spec.js

```diff
@@ -1,180 +1,179 @@
-describe('Class hierarchy screen validation', function () {
-    let repoId = 'repo' + Date.now();
+const INITIAL_CLASS_COUNT = 50;
+const SEARCH_INPUT_DROPDOWN_ID = '#search_input_dropdown';
+const CLASS_LABEL_SELECTOR = '#main-group > text.label';
+
+describe('Class hierarchy screen validation', () => {
+    let repositoryId;
+
+    before(() => {
+        repositoryId = 'repo' + Date.now();
+        cy.createRepository({id: repositoryId});
 
-    before(function () {
         cy.visit('/repository');
-        // Create new one
-        cy.createNewRepo(repoId).wait(2000);
-        cy.setRepoDefault(repoId);
-        cy.selectRepo(repoId);
-
-        // Import Wine dataset which will be used in most of the following tests
-        cy.navigateToPage('Import', 'RDF');
-        cy.openImportURLDialog('https://www.w3.org/TR/owl-guide/wine.rdf');
-        cy.clickImportUrlBtn();
-        cy.clickImportBtnOnPopUpMenu().wait(2000);
+        cy.setRepoDefault(repositoryId);
+        cy.selectRepo(repositoryId);
+
+        cy.importFromUrl(repositoryId, 'https://www.w3.org/TR/owl-guide/wine.rdf', {});
     });
 
-    beforeEach(function () {
-        // Go to Explore\Class Hierarchy
+    beforeEach(() => {
         cy.visit('/hierarchy').wait(1000);
     });
 
-    after(function () {
-        cy.navigateToPage('Setup', 'Repositories');
-        cy.deleteRepo(repoId);
+    after(() => {
+        cy.deleteRepository(repositoryId);
     });
 
-    it('Test initial state of the diagram has a class count 50', function () {
-        //  Confirm the initial state of the diagram has a class count of 50
-        cy.get('.rz-pointer[role="slider"]')
-            .then(($element) => {
-                expect($element.prop('aria-valuenow') === 50);
-            });
-    });
-
-    it('Test show/hide prefixes', function () {
-        //  Verify that switching on/off Show/hide prefixes is reflected on the diagram - prefixes are displayed/hidden
-        cy.get('#main-group > text.label')
-            .each(($element) => {
-                // let display = ;
-                let parsed = $element.prop('style');
-                if (parsed.display != 'none') {
-                    expect($element.text()).contain(':');
-                }
+    function getCurrentSliderValue() {
+        // The count is taken from the rz-pointer's attribute and not from a visible in the UI value
+        // as the rz-slider library doesn't provide a reliable way to get this. It just has multiple
+        // '.rz-bubble' elements and no appropriate selector for the one which holds the visible
+        // value.
+        return cy.get('.rz-pointer[role="slider"]')
+            .then(($element) => $element.prop('aria-valuenow'));
+    }
+
+    function searchForClass(name) {
+        cy.get('.toolbar-holder .icon-search')
+            .click()
+            .then(() => {
+                cy.get('#search_input_value').type(name).type('{enter}').wait(500);
```

Comment:
Document the need of the `wait` or try to properly wait?

Reply:
I actually have forgotten about this. I'll just try to refactor the `findClassByName` function and remove the wait. Currently the way the class is searched in the diagram is ugly and would cause the tests to fail if the wait is missing because cypress doesn't wait in the `each` loop as I can remember.

Reply:
Actually I can't refactor it because of the way diagram works. Visible classes visualized with circles and their labels are not in any relation with each other. Labels (the text elements) are not descendants of the circles in the DOM tree. The labels of some classes on the other side might be clipped in order to not overlap with other labels if their circle is too small. So the only way to get a class label is to get it through the data property of the element which makes it impossible to craft a simple css selector to grab the desired text element.